relayer/application.ts

```diff
@@ -212,25 +276,35 @@ export class RelayerApp<ContextT extends Context> extends EventEmitter {
    * @param vaa
    * @param opts
    */
-  async pushVaaThroughPipeline(vaa: Buffer, opts?: any): Promise<void> {
-    const parsedVaa = wormholeSdk.parseVaa(vaa);
+  private async pushVaaThroughPipeline(
+    vaa: SignedVaa,
+    opts: any,
+  ): Promise<void> {
+    const parsedVaa = parseVaaWithBytes(vaa);
+    const job: RelayJob | undefined = opts.job;
+
     let ctx: Context = {
-      vaa: parsedVaa,
-      vaaBytes: vaa,
-      env: this.env,
-      fetchVaa: this.fetchVaa.bind(this),
-      fetchVaas: this.fetchVaas.bind(this),
-      processVaa: this.processVaa.bind(this),
       config: {
         spyFilters: await this.spyFilters(),
       },
+      env: this.env,
+      fetchVaa: this.fetchVaa.bind(this),
+      fetchVaas: this.fetchVaas.bind(this),
       locals: {},
+      on: this.on.bind(this),
+      processVaa: this.processVaa.bind(this),
+      vaa: parsedVaa,
+      vaaBytes: vaa,
     };
-    Object.assign(ctx, opts);
+    Object.assign(ctx, opts, { storage: { job } });
```

Comment:
This isn't a mergeDeep so this will overwrite any key with storage already present. What's the rationale behind putting this under the storage key?

Reply:
it was in the storage key in the beginning. This was being done elsewhere and i moved it here but I think I can move it back where it used to live.